ironfish-cli/src/utils/currency.ts

```diff
@@ -29,17 +30,53 @@ export async function promptCurrency(options: {
     account?: string
     assetId?: string
     confirmations?: number
+    minimalBalance?: bigint
   }
 }): Promise<bigint | null> {
   let text = options.text
+  let balanceInOre = 0n
+  let balance = 0n
 
   if (options.balance) {
-    const balance = await options.client.getAccountBalance({
+    const balanceResponse = await options.client.getAccountBalance({
       assetId: options.balance.assetId ?? Asset.nativeId().toString('hex'),
       confirmations: options.balance.confirmations,
     })
 
-    text += ` (balance ${CurrencyUtils.renderIron(balance.content.available)})`
+    balance = CurrencyUtils.decode(balanceResponse.content.available)
+
+    if (options.balance.assetId === Asset.nativeId().toString('hex')) {
+      balanceInOre = balance
+    }
+
+    if (options.balance.minimalBalance && balance < options.balance.minimalBalance) {
+      if (options.balance.minimalBalance === 2n) {
+        options.logger.log(
+          'Balance is not enough for the transaction. Require a minimal transaction of 1 ore and minimal transaction fee of 1 ore.',
+        )
+      } else {
+        options.logger.log('Balance is not enough for the transaction.')
+      }
+
+      return null
+    }
+
+    text += ` (balance ${CurrencyUtils.renderIron(balance)})`
+  }
+
+  // get balance of native asset if above code have not got it.
+  if (balanceInOre === 0n) {
+    const balanceResponse = await options.client.getAccountBalance({
+      assetId: Asset.nativeId().toString('hex'),
+      confirmations: options.balance?.confirmations ?? 0,
+    })
+
+    balanceInOre = CurrencyUtils.decode(balanceResponse.content.available)
+
+    if (balanceInOre <= 0) {
+      options.logger.log('Balance of native asset is not enough for the transaction.')
```

Comment:
I'm not sure this check belongs in the prompt. If we want to make sure that the user has a non-zero IRON balance before trying to create any transaction then I think checking in the same place that we do for the node's sync status might be better.

If the user passes the amount and/or fee in through flags then this check would never be evaluated.